app/src/main/java/com/example/tasky/agenda_details/presentation/AgendaDetailsViewModel.kt

```diff
@@ -317,6 +326,11 @@ class AgendaDetailsViewModel @Inject constructor(
                     }
                 }
             }
+
+            withContext(Dispatchers.IO) {
+                // delete in local db
+                eventLocalRepository.deleteEvent(eventId = eventId)
+            }
```

Comment:
The switch here is not necessary. The responsibility to switch to the right dispatcher lies at the function where the blocking call gets converted to the suspending function. In this case, somewhere deep down in the Room library there will be a blocking (and not suspending) call which wraps the blocking call with `withContext(Dispatchers.IO)`.

Everything else would be super error-prone since not remembering to switch to the right dispatcher would always result in worse performance or bugs.